File: KahootWebApi/ViewModels/LoginModel.cs
using System.ComponentModel.DataAnnotations;

namespace KahootWebApi.ViewModels
{
    public class LoginModel
    {
        [Required(ErrorMessage = "Не указан UserName")]
        public string? UserName { get; set; }

        [Required(ErrorMessage = "Не указан пароль")]
        [DataType(DataType.Password)]
        public string? Password { get; set; }
    }
}

File: KahootClient/src/LoginModel.ts
export class LoginModel{
  public id: number | undefined;

  constructor(public userName: string | undefined,
              public password: string | undefined) {  }

  public get Id() {
    return this.id;
  }

  public get Username() {
    return this.userName;
  }

  public get Password() {
    return this.password;
  }
}
